File: Chapter 05/C Code/5_10.c
#include <stdio.h>

int main() {
    int a[3][3] = {{1,1,1},{0,1,1},{0,0,1}};
    int flag = 0;
    
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < i; j++) {
            if (a[i][j] != 0) {
                flag = 1;
                break;
            }
        }
        if (flag == 1) {
            break;
        }
    }
    
    if (flag == 0) {
        printf("Matrix is upper triangular\n");
    } else {
        printf("Matrix is not upper triangular\n");
    }
    
    return 0;
}

File: Chapter 05/Python code/5_10.py
a=[[1,1,1],[10,11,12],[0,0,1]]

flag=0
for i in range(0, 3):
    for j in range(0, i):
        if a[i][j]!=0:
            flag=1
            break

print("Matrix is ", a)
if flag==0:
    print("Matrix is upper triangular ")
else:
    print("Matrix is not upper triangular ")
